dip-0008.md

```diff
@@ -275,6 +277,127 @@ resolve any ambiguities which might occur in the last few blocks.
+ChainLock might depend on the attackers motivations.  An attacker with
+control of a ChainLock quorum would be able to:
+1. Produce conflicting ChainLocks.
+2. Withhold ChainLocks and then issue a ChainLock for a conflicting previous block
+not included in the current chain of most work.
+3. Secure undo advantage.
+
+We assume that all masternodes not under malicious control are
+behaving as expected. In practice this is generally observed.  In theory,
+it is possible for some nodes to be unreachable, or the attacker could
+combine a DoS attack in conjuction with compromising nodes. If this is the case, it
+would become more likely an attacker could withhold ChainLocks, and not
+be any more likely the attacker could produce a malicious ChainLock.
+Our calculations can easily be modified to model the network operating
+under other assumptions.
+
+In this section we will show that the random selection of a LLMQ will
+not allow an attacker to control a ChainLock quorum without control of a
+supermajority of masternodes.
+
+Quorum selection is designed to provide for an effectively simple random
+choice of masternodes in each quorum. Thus we may assume that each quorum is
+a simple random selection. If there are a total of 'N' masternodes the
+number of possible quorums is the number of ways that 400 objects can be
+chosen out of `N`. In some literature this number is
+written <sub>`N`</sub>`C`<sub>`400`</sub> and is enunciated as
+'N choose 400'. Numbers that arise in this fashion are all
+binomial coefficients and have a delightful relation to Pascal's Triangle.
+
+For any process with an observable outcome the set of all outcomes is
+called a sample space.  If all outcomes are equally likely, as in the
+selection of a quorum, the probability of any event is the number of
+outcomes that satisfy that event over the total number outcomes.
+
+An attacker can temporarily prevent ChainLocks if they control
+`161` out of 400 masternodes in the quorum.  If the attacker controls
+`m` out of the `N` total masternodes we can compute the
+number of outcomes that satisfy the event "exactly `161`
+nodes of the quorum are controlled by the masternode."
+Specifically there are
+<sub>`m`</sub>`C`<sub>`161`</sub> * <sub>`N - m`</sub>`C`<sub>`239`</sub>
+ outcomes where the attacker controls exactly `161` masternodes.
+Thus, the probability that the attacker controls exactly `161` nodes is
+
+<p align="center"><img src="dip-0008/ThreshCase.gif" alt="Threshold Probability"></p>
+
+The attacker could also be successful if they control <b>more</b> than `160`
+of the nodes in the quorum selection. Thus the exact probability that an
+attacker can successfully prevent ChainLocks for that quorum round is
+computed by counting the number of outcomes where the attacker controls
+160 or more nodes in the quorum divided by the total number of outcomes.
+This is computed by the sum
+
+<p align="center"><img src="dip-0008/fin_sum.gif" alt="Final sum" ></p>
```

Comment:
Can't comment on the gif itself but it looks like the right top part of the equation is "sub" when it shouldn't be:
in the gif: <sub>`m`</sub>`C`<sub>`160 + j` * <sub>`N - m`</sub>`C`<sub>`240 - j`</sub></sub> 
should be: <sub>`m`</sub>`C`<sub>`160 + j`</sub> * <sub>`N - m`</sub>`C`<sub>`240 - j`</sub>

Reply:
The comment is correct.  The C should not be a subscript.

Reply:
Corrected in cf01b58